### PR TITLE
TOML config: validate auth options

### PR DESCRIPTION
**Pull request by chrzaszcz**
Test and validate auth options.

Also: remove the confusing "cyrsasl_*" prefix.

### CODE REVIEW

**Review by NelsonVides — APPROVED** [review]
Seems very good to me 🙂

File: src/config/mongoose_config_validator_toml.erl [hunk 1]
```diff
@@ -145,6 +145,35 @@ validate([item, _TypeBin, <<"handlers">>, item, <<"http">>, <<"listen">>],
...
+%% auth
+validate([item, <<"methods">>, <<"auth">>],
+         [Value]) ->
+    validate_module(list_to_atom("ejabberd_auth_" ++ atom_to_list(Value)));
+validate([<<"password">>, <<"auth">>],
+         [{password_format, Value}]) ->
+    validate_password_format(Value);
+validate([item, <<"hash">>, <<"password">>, <<"auth">>],
+         [Value]) ->
+    validate_enum(Value, [sha, sha224, sha256, sha384, sha512]);
```
[the remark this answers is not in the document]
Comment (chrzaszcz) [reply src/config/mongoose_config_validator_toml.erl:157]: You're right, actually I would validate this in a better way if it existed. Unfortunately, there is no easy check here, it needs to be added to `mongoose_scram` and I did not want to do it in this PR.

**Review by kmakiela — APPROVED** [review]
Great work, will have to remember to test and validate currently missing `ldap`options that will come in a PR soon

**Comment by kmakiela** [comment]
[This PR](https://github.com/esl/MongooseIM/pull/2852) expands `auth` section

**Pull request merged**